src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -1372,7 +1384,12 @@ def forward(
                 src_flatten = hidden_states[enc_ind].flatten(2).permute(0, 2, 1)
                 if self.training or self.eval_size is None:
                     pos_embed = self.build_2d_sincos_position_embedding(
-                        width, height, self.encoder_hidden_dim, self.positional_encoding_temperature
+                        width,
+                        height,
+                        self.encoder_hidden_dim,
+                        self.positional_encoding_temperature,
+                        device=src_flatten.device,
+                        dtype=src_flatten.dtype,
                     ).to(src_flatten.device, src_flatten.dtype)
```

Comment:
I don't yhink we need `.to()` if we give device and dtype into the function

Reply:
Indeed thanks!